plugins/package/src/main/kotlin/net/corda/cli/plugins/packaging/CreateCpb.kt

```diff
@@ -0,0 +1,130 @@
+
+@Command(
+    name = "create-cpb",
+    description = ["Creates a CPB from passed in CPK archives."]
+)
+class CreateCpb : Runnable {
+
+    @Option(names = ["--cpks"], required = true, split = " ", description = ["CPKs to convert into CPB"])
+    lateinit var cpks: List<String>
+
+    @Option(names = ["--cpb-name"], required = true, description = ["Manifest attribute CPB name"])
+    lateinit var cpbName: String
+
+    @Option(names = ["--cpb-version"], required = true, description = ["Manifest attribute CPB version"])
+    lateinit var cpbVersion: String
+
+    // Is the below option required? If not what should be its default value?
+    @Option(names = ["--upgrade"], required = true, description = ["CPB Upgrade option"])
+    lateinit var cpbUpgrade: String
+
+    @Option(names = ["--file", "-f"], required = true, description = ["Output CPB file name"])
+    lateinit var outputCpbFileName: String
+
+    @Option(names = ["--keystore", "-s"], required = true, description = ["Keystore holding signing keys"])
+    lateinit var keyStoreFileName: String
+
+    @Option(names = ["--storepass", "--password", "-p"], required = true, description = ["Keystore password"])
+    lateinit var keyStorePass: String
+
+    @Option(names = ["--key", "-k"], required = true, description = ["Key alias"])
+    lateinit var keyAlias: String
+
+    @Option(names = ["--tsa", "-t"], description = ["Time Stamping Authority (TSA) URL"])
+    var tsaUrl: String? = null
+
+    companion object {
+        private val CPB_NAME_ATTRIBUTE = Attributes.Name("Corda-CPB-Name")
+
+        private val CPB_VERSION_ATTRIBUTE = Attributes.Name("Corda-CPB-Version")
+
+        private val CPB_FORMAT_VERSION = Attributes.Name("Corda-CPB-Format")
+
+        const val CPB_CURRENT_FORMAT_VERSION = "2.0"
+
+        private val MANIFEST_CPB_UPGRADE = Attributes.Name("Corda-CPB-Upgrade")
+
+        const val CPB_FILE_EXTENSION = "cpb"
+
+        /**
+         * Name of signature within Cpb file
+         */
+        private const val CPB_SIGNER_NAME = "CPB-SIG"
+
+        /**
+         * Check file exists and returns a Path object pointing to the file, throws error if file does not exist
+         */
+        private fun requireFileExists(fileName: String): Path {
+            val path = Path.of(fileName)
+            require(Files.isReadable(path)) { "\"$fileName\" does not exist or is not readable" }
+            return path
+        }
+
+        /**
+         * Check that file does not exist and returns a Path object pointing to the filename, throws error if file exists
+         */
+        private fun requireFileDoesNotExist(fileName: String): Path {
+            val path = Path.of(fileName)
+            require(Files.notExists(path)) { "\"$fileName\" already exists" }
+            return path
+        }
+    }
+
+    override fun run() {
+        val unsignedCpb = Files.createTempFile("temp-unsigned-cpb-${UUID.randomUUID()}", ".$CPB_FILE_EXTENSION")
```

Comment:
We don't need the random UUID here. `createTempFile` will give us an unused filename.